### PR TITLE
Cache several char[] allocations in System.Private.Xml library methods

### PR DESCRIPTION
**Pull request by TrayanZapryanov**
I've found these allocations while looking for new char[]. Changes looks safe.

- Cache creation of char[] used in System.Xml.Extensions.ExtensionMethods.ToUri
- Cache creation of char[] used in System.Xml.Serialization.CodeIdentifier.EscapeKeywords

### CODE REVIEW

File: src/libraries/System.Private.Xml/src/System/Xml/Extensions/ExtensionMethods.cs
Comment (stephentoub) [inline src/libraries/System.Private.Xml/src/System/Xml/Extensions/ExtensionMethods.cs:8]:
```suggestion
        private static readonly char[] s_trimChars = new char[] { ' ', '\t', '\n', '\r' };

```

File: src/libraries/System.Private.Xml/src/System/Xml/Serialization/CodeIdentifier.cs
Comment (stephentoub) [inline src/libraries/System.Private.Xml/src/System/Xml/Serialization/CodeIdentifier.cs:274]:
```suggestion
        private static readonly char[] s_identifierSeparators = new char[] {'.', ',', '<', '>'};

```

File: src/libraries/System.Private.Xml/src/System/Xml/Serialization/CodeIdentifier.cs
Comment (stephentoub) [inline src/libraries/System.Private.Xml/src/System/Xml/Serialization/CodeIdentifier.cs:276]: This function is suboptimal in a variety of ways.  If it's on a hot path, there are other improvements that can be made.

**Review by stephentoub — APPROVED** [review]

File: src/libraries/System.Private.Xml/src/System/Xml/Extensions/ExtensionMethods.cs
Comment (TrayanZapryanov) [reply src/libraries/System.Private.Xml/src/System/Xml/Extensions/ExtensionMethods.cs:8]: @krwq I've checked once more and this function looks same as XmlConvert.ToUri. Maybe it is better to delete this and use other one instead of optimizing it ?

File: src/libraries/System.Private.Xml/src/System/Xml/Serialization/CodeIdentifier.cs
Comment (TrayanZapryanov) [reply src/libraries/System.Private.Xml/src/System/Xml/Serialization/CodeIdentifier.cs:276]: Maybe @krwq can say if he think that this is a hot path, I check code and see that it is used in several places also have several others which are quite similar and can be optimized. If we receive approval that it is hot - I can prepare another PR with my suggestions for improvements.

File: src/libraries/System.Private.Xml/src/System/Xml/Extensions/ExtensionMethods.cs
Comment (krwq) [reply src/libraries/System.Private.Xml/src/System/Xml/Extensions/ExtensionMethods.cs:8]: if you think they're exactly the same that sounds like a good idea.

File: src/libraries/System.Private.Xml/src/System/Xml/Serialization/CodeIdentifier.cs
Comment (krwq) [reply src/libraries/System.Private.Xml/src/System/Xml/Serialization/CodeIdentifier.cs:277]: I'm not super familiar with serialization part of the code but quick search looks like it's only used to generate C# code which is most likely a single time cost overall which doesn't sound like a hot path to me. It's not bad idea to optimize it anyway but I'd prefer to leave it for separate PR.

**Review by krwq — APPROVED** [review]
Thanks! Consider doing the suggestion from your comment. Let me know if you plan to leave as is (I'll wait with merging until I hear from you or merge if not hear in couple of days).

File: src/libraries/System.Private.Xml/src/System/Xml/Extensions/ExtensionMethods.cs
Comment (TrayanZapryanov) [reply src/libraries/System.Private.Xml/src/System/Xml/Extensions/ExtensionMethods.cs:8]: I will remove it as I don't see any difference:
` private static readonly char[] s_trimChars = new char[] { ' ', '\t', '\n', '\r' };

        internal static Uri ToUri(string s)
        {
            if (s != null && s.Length > 0)
            { //string.Empty is a valid uri but not "   "
                s = s.Trim(s_trimChars);
                if (s.Length == 0 || s.IndexOf("##", StringComparison.Ordinal) != -1)
                {
                    throw new FormatException(SR.Format(SR.XmlConvert_BadFormat, s, "Uri"));
                }
            }
            if (!Uri.TryCreate(s, UriKind.RelativeOrAbsolute, out Uri? uri))
            {
                throw new FormatException(SR.Format(SR.XmlConvert_BadFormat, s, "Uri"));
            }

            return uri;
        }`

and

internal static Exception? TryToUri(string s, out Uri? result)
        {
            result = null;

            if (s != null && s.Length > 0)
            { //string.Empty is a valid uri but not "   "
                s = TrimString(s);
                if (s.Length == 0 || s.IndexOf("##", StringComparison.Ordinal) != -1)
                {
                    return new FormatException(SR.Format(SR.XmlConvert_BadFormat, s, "Uri"));
                }
            }
            if (!Uri.TryCreate(s, UriKind.RelativeOrAbsolute, out result))
            {
                return new FormatException(SR.Format(SR.XmlConvert_BadFormat, s, "Uri"));
            }

            return null;
        }

File: src/libraries/System.Private.Xml/src/System.Private.Xml.csproj
Comment (krwq) [inline src/libraries/System.Private.Xml/src/System.Private.Xml.csproj:793]: is this actually needed?

File: src/libraries/System.Private.Xml/src/System/Xml/Serialization/XmlSerializerNamespaces.cs
Comment (krwq) [inline src/libraries/System.Private.Xml/src/System/Xml/Serialization/XmlSerializerNamespaces.cs:71]: I'll recommend to remove the comment since the behavior is identical as it was

**Review by krwq — APPROVED** [review]
LGTM after fixing/resolving last two comments

**Pull request merged**
